contracts/pool/pools/liquid/WithdrawalEligibility.sol

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.18;
+
+import "../TokenPoolBase.sol";
+
+error UnlockPeriodNotFinished();
+error NonWithdrawableTokens();
+error AmountExceedsAvailableWithdraw();
+
+/**
+ * @title WithdrawalEligibility
+ * @dev Provides functionality for checking withdrawal eligibility and grace periods
+ */
+contract WithdrawalEligibility is TokenPoolBase {
+    uint256 internal constant GRACE_PERIOD = 4 weeks;
+    uint256 internal constant GRACE_PERIOD_REMOVAL_FEE = 1000;
+    uint256 internal constant UPGRADE_PERIOD = 63_158_400; // 2 years considering leap years in seconds;
+
+    struct EligbleWithdraw {
+        uint256 gracePeriodEndTime;
+        uint256 amount;
+    }
+
+    struct Stake {
+        EligbleWithdraw eligbleWithdraw;
+        uint256 amount;
+    }
+
+    struct Lock {
+        EligbleWithdraw eligbleWithdraw;
+        uint256 lockTime;
+        uint256 amount;
+    }
+
+    mapping(address => Stake) internal stakes;
```

Comment:
Mozemo li dodat imena ovim parametrima u mappingu i na ostale mappinge da bude malo citljivije

```suggestion
    mapping(address user => Stake stake) internal stakes;
```

Reply:
mogu ili samo da dodam komentar poviše mappinga koji opisuje sta se mapira na sta. Mislim tako bi inace radili

Reply:
Prije verzije 0.8.18 se nije moglo dodat ime parametrima u mappingu, a posto se sad to moze mislim da je bolje stavit ime direktno. Vecinom smo i davali imena kad sam bacia oko na cili repo.

https://github.com/Endemic-NFT/ENDEMIC/blob/2ea720113d2824ac5895032efafb5a68de6858e8/contracts/manager/PaymentManager.sol#L13-L14

https://github.com/Endemic-NFT/ENDEMIC/blob/2ea720113d2824ac5895032efafb5a68de6858e8/contracts/exchange/mixins/EndemicNonceManager.sol#L5

Reply:
moze moze dodam to